src/scripts/fabric.js

```diff
@@ -1,6 +1,107 @@
 import * as fabric from "fabric";
 import Engine from "./engine";
 
+// fabric.StaticCanvas.prototype.renderCanvas = function (ctx, objects) {
+//   this.clearContext(this.getContext());
+//   this._renderObjects(ctx, objects);
+// };
+
+const uAffine = (product, curr) =>
+  curr ? fabric.util.multiplyTransformMatrices(curr, product, false) : product;
+
+const u = (product, curr) =>
+  curr ? fabric.util.multiplyTransformMatrices(curr, product, true) : product;
+
+const multiplyTransformMatrixArray = (matrices, is2x2) =>
+  matrices.reduceRight(is2x2 ? u : uAffine, fabric.iMatrix.concat());
+
+class OptimizedRect extends fabric.Rect {
+  m;
+  calcOwnMatrix() {
+    // return [1, 0, 0, 1, this.left, this.top];
```

Comment:
to test the perf of this impl of `calcOwnMatrix` try to uncomment this line and compare results. Should be the same